### ISSUE TITLE
implement navigation hacks for climbing above

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#1867]
I had a PoC for a hack that allows bots to get limited climbing abilities.
So far, the PoC I had (long ago, almost a year old) showed to be able to:

* make humans climb a ladder
* make wall-walking aliens climb a wall

However, I believe this can be used for at least marauders, since they do climp too.
For dragoons and adv goons, there might be some hope, and obviously, tyrants are just a lost cause.

All this relies on unclean code, workarounds and manual building of connections in maps, though.
I don't even still have a working code for this, only old videos as a proof it is possible.

**Comment by illwieckz** [comment]
Quick&dirty patch to print some ladder coordinates that may be good enough to generate navcons:

```diff
diff --git a/src/shared/navgen/nav.cpp b/src/shared/navgen/nav.cpp
index 4ff58fdf3..ad1b48472 100644
--- a/src/shared/navgen/nav.cpp
+++ b/src/shared/navgen/nav.cpp
@@ -305,6 +305,27 @@ void NavmeshGenerator::LoadTris( std::vector<float> &verts, std::vector<int> &tr
                                FixWinding( w );
                        }
 
+                       if ( brushShader->surfaceFlags & SURF_LADDER ) {
+                               Log::Warn("found ladder surface!");
+                               if ( w )
+                               {
+                                       for ( int j = 0; j < w->numpoints - 2; j++ ) {
+                                               float *v1 = w->p[j];
+                                               float *v2 = w->p[j + 1];
+                                               float *v3 = w->p[j + 2];
+                                               Log::Warn(
+                                                       "found ladder point: "
+                                                       "%d %d %d, "
+                                                       "%d %d %d, "
+                                                       "%d %d %d",
+                                                       v1[0], v1[1], v1[2],
+                                                       v2[0], v2[1], v2[2],
+                                                       v3[0], v3[1], v3[2]
+                                                       );
+                                       }
+                               }
+                       }
+
                        if ( w ) {
                                for ( int j = 2; j < w->numpoints; j++ ) {
                                        AddTri( verts, tris, w->p[0], w->p[j - 1], w->p[j] );
```

**Comment by bmorel** [comment]
thanks, this will be useful to get automatic handling, instead of manually placing them.

**Comment by bmorel** [comment]
partially fixed by #2566 but still need wallwalking.

**Comment by bmorel** [comment]
@sweet235 implemented wallwalking.
